src/hist1d.jl

```diff
@@ -98,9 +98,15 @@ end
     r = @inbounds h.hist.edges[1]
     L = length(r) - 1
     binidx = _edge_binindex(r, val)
-    if unsigned(binidx - 1) < L
+    if h.overflow
```

Comment:
does this get constant folded? If not I would multiply weight and sumw2 by `h.overflow` and always clamp to last bin to eliminate one branch

Reply:
When overflow=false, we get identical performance as before, so this is optimized. When overflow=true, it's only ~10% slower due to the clamp.